pulp_ansible/tests/functional/api/collection/v3/test_collection_naming_edgecases.py

```diff
@@ -0,0 +1,336 @@
+                        cv_info = pulp_client.get(cv_summary["href"])
+
+                        # include the repoid & distid ...
+                        cv_info["repository_id"] = repos[repo_name]["pulp_href"].split("/")[-2]
+                        if repo_name in dists:
+                            did = dists[repo_name]["pulp_href"]
+                            did = did.split("/")[-2]
+                            cv_info["distribution_id"] = did
+                            cv_info["base_path"] = dists[repo_name]["base_path"]
+                        else:
+                            cv_info["distribution_id"] = None
+                            cv_info["base_path"] = None
+
+                        key = (
+                            repo_name,
+                            cv_info["namespace"]["name"],
+                            cv_info["name"],
+                            cv_info["version"],
+                        )
+                        cvs[key] = cv_info
+
+    return cvs
+
+
+def get_collection_versions(pulp_client):
+    cversions = pulp_client.get("/pulp/api/v3/content/ansible/collection_versions/")
+    return dict(((x["namespace"], x["name"], x["version"]), x) for x in cversions)
+
+
+def get_repositories(pulp_client):
+    repos = pulp_client.get(ANSIBLE_REPO_PATH)
+    repos = dict((x["name"], x) for x in repos)
+    return repos
+
+
+def get_distributions(pulp_client):
+    dists = pulp_client.get(ANSIBLE_DISTRIBUTION_PATH)
+    dists = dict((x["name"], x) for x in dists)
+    return dists
+
+
+@pytest.fixture(scope="session")
+def pulp_repo(pulp_client):
+    """Find or create a Repository to attach to the Ansible Distribution we create."""
+    repos = pulp_client.get(ANSIBLE_REPO_PATH)
+    if repos:
+        yield repos[0]
+    else:
+        repo_data = gen_repo(name="automation-hub")
+        repo = pulp_client.post(ANSIBLE_REPO_PATH, repo_data)
+        yield repo
+        pulp_client.delete(repo["pulp_href"])
+
+
+@pytest.fixture(scope="session")
+def pulp_dist(pulp_client, pulp_repo):
+    """Create an Ansible Distribution to simulate the automation hub environment for testing."""
+    dists = pulp_client.get(ANSIBLE_DISTRIBUTION_PATH + "?base_path=automation-hub")
+
+    if len(dists) == 0:
+        dist_data = gen_distribution(
+            name="automation-hub", base_path="automation-hub", repository=pulp_repo["pulp_href"]
+        )
+        dist = pulp_client.post(ANSIBLE_DISTRIBUTION_PATH, dist_data)
+        created = True
+    elif len(dists) == 1:
+        dist = dists[0]
+        created = False
+    else:
+        raise ValueError("Found too many Ansible Distributions at 'automation-hub'.")
+    yield dist
+    if created:
+        pulp_client.delete(dist["pulp_href"])
+
+
+@pytest.mark.pulp_on_localhost
+def test_collection_named_collections(pulp_client, tmp_path, ansible_repo_api_client, pulp_dist):
+    # https://issues.redhat.com/browse/AAH-2158
+    delete_orphans()
```

Comment:
Do we need this?
Can we instead randomize the namespace? It may allow to mark the test as parallel.